src/preamble.js

```diff
@@ -164,42 +171,41 @@ function ccall (ident, returnType, argTypes, args, opts) {
     }
   }
   var ret = func.apply(null, cArgs);
-#if ASSERTIONS
 #if EMTERPRETIFY_ASYNC
-  if ((!opts || !opts.async) && typeof EmterpreterAsync === 'object') {
-    assert(!EmterpreterAsync.state, 'cannot start async op with normal JS calling ccall');
-  }
-  if (opts && opts.async) assert(!returnType, 'async ccalls cannot return values');
+  if (typeof EmterpreterAsync === 'object' && EmterpreterAsync.state) {
```

Comment:
please add a comment here explaining the case of `EmterpreterAsync.state` being 0 when the user specified async, and that we create a promise later down if so, so the result is always a promise in that case

Reply:
Good point, thanks.